### PR TITLE
Change SDK name of Sentry Native for Flutter

### LINKED ISSUE #1333: Change SDK name of Sentry Native for Flutter
### Description

Right now it is `sentry.native.android` and this is set at compile time.
Ideally, this would be `sentry.native.android.flutter` or just `sentry.native.flutter`.

### PR DESCRIPTION
**Pull request by denrase**
#skip-changelog

## :scroll: Description

Update SDK names set in native plugins.

## :bulb: Motivation and Context

Closes  #1333

## :pencil: Checklist

- [ ] I reviewed submitted code
- [ ] I added tests to verify changes
- [ ] No new PII added or SDK only sends newly added PII if `sendDefaultPii` is enabled
- [ ] I updated the docs if needed
- [ ] All tests passing
- [ ] No breaking changes

### CODE REVIEW

File: flutter/android/src/main/kotlin/io/sentry/flutter/SentryFlutterPlugin.kt [hunk 1]
```diff
@@ -41,7 +41,7 @@ class SentryFlutterPlugin : FlutterPlugin, MethodCallHandler, ActivityAware {
   private var autoPerformanceTracingEnabled = false
 
   private val flutterSdk = "sentry.dart.flutter"
-  private val androidSdk = "sentry.java.android.flutter"
+  private val androidSdk = "sentry.native.android.flutter"
```
Comment (marandaneto) [inline flutter/android/src/main/kotlin/io/sentry/flutter/SentryFlutterPlugin.kt]: This change isn't needed.
The changes are about the `nativeSdk`

File: flutter/ios/Classes/SentryFlutterPluginApple.swift [hunk 2]
```diff
@@ -12,7 +12,7 @@ import AppKit
 // swiftlint:disable:next type_body_length
 public class SentryFlutterPluginApple: NSObject, FlutterPlugin {
 
-    private static let nativeClientName = "sentry.cocoa.flutter"
+    private static let nativeClientName = "sentry.native.cocoa.flutter"
```
Comment (marandaneto) [inline flutter/ios/Classes/SentryFlutterPluginApple.swift]: This change isn't needed, the issue is only about the Sentry Native for Android NDK

**Comment by denrase** [comment]
@marandaneto I tried to check the SDK names on sentry.io through the example app, but i'm not seeing those values set, neither for iOS nor Android. Am I looking in the wrong place?

Bildschirmfoto 2023-05-22 um 11 10 07

File: flutter/ios/Classes/SentryFlutterPluginApple.swift (hunk 2 above)
Comment (denrase) [reply flutter/ios/Classes/SentryFlutterPluginApple.swift]: So we don't need to align on the coco side that the naming is the same?

**Comment by marandaneto** [comment]
> @marandaneto I tried to check the SDK names on sentry.io through the example app, but i'm not seeing those values set, neither for iOS nor Android. Am I looking in the wrong place?
> 
> Bildschirmfoto 2023-05-22 um 11 10 07

The place is right, but events captured on the Dart layer will be this name.
See the `AndroidExample` class in `main.dart`.

When this method is executed https://github.com/getsentry/sentry-dart/blob/d883d62345e5c21066faf14885eea7afd75b89e1/flutter/example/android/app/src/main/kotlin/io/sentry/samples/flutter/MainActivity.kt#L30
SDK name should be `sentry.java.android.flutter` (it is already).

When those 2 methods are called https://github.com/getsentry/sentry-dart/blob/d883d62345e5c21066faf14885eea7afd75b89e1/flutter/example/android/app/src/main/kotlin/io/sentry/samples/flutter/MainActivity.kt#L33-L38
SDK name it should be called `sentry.native.android.flutter` -> this is the one that needs fixing.

File: flutter/ios/Classes/SentryFlutterPluginApple.swift (hunk 2 above)
Comment (marandaneto) [reply flutter/ios/Classes/SentryFlutterPluginApple.swift]: Sentry the confusion, when I meant Sentry Native isn't about Android and iOS, its about https://github.com/getsentry/sentry-native
SentryNative for NDK support.- Android.

**Comment by marandaneto** [comment]
@denrase example https://github.com/getsentry/sentry-react-native/pull/3086/files

**Review by marandaneto — APPROVED** [review]

**Pull request merged**